zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -78,6 +83,58 @@ use input_selection::{
     GreedyInputSelector, GreedyInputSelectorError, InputSelector, InputSelectorError,
 };
 
+/// Represents the identifier for an account that was derived from a ZIP-32 HD seed and account index.
+#[derive(Debug, Clone)]
+pub struct HDSeedAccount(
+    pub HDSeedFingerprint,
+    pub zip32::AccountId,
+    pub UnifiedFullViewingKey,
+);
+
+/// Represents an arbitrary account for which the seed and ZIP-32 account ID are not known
+/// and may not have been involved in creating this account.
+#[derive(Debug, Clone)]
+pub enum ImportedAccount {
+    /// An account that was imported via its full viewing key.
+    Full(UnifiedFullViewingKey),
+}
+
+/// The inputs for adding an account to a wallet.
+#[derive(Debug, Clone)]
+pub enum Account {
+    /// Inputs for a ZIP-32 HD account.
+    Zip32(HDSeedAccount),
+    /// Inputs for an imported account.
+    Imported(ImportedAccount),
+}
+
+impl Account {
+    /// Gets the default UA for the account.
+    pub fn default_address(
+        &self,
+        request: UnifiedAddressRequest,
+    ) -> (UnifiedAddress, DiversifierIndex) {
+        match self {
+            Account::Zip32(HDSeedAccount(_, _, ufvk)) => ufvk.default_address(request),
+            Account::Imported(ImportedAccount::Full(ufvk)) => ufvk.default_address(request),
+        }
+    }
+
+    /// Gets the unified full viewing key for this account, if available.
+    ///
+    /// Accounts initialized with an incoming viewing key will not have a unified full viewing key.
+    pub fn ufvk(&self) -> Option<&UnifiedFullViewingKey> {
```

Comment:
The inclusion of this method and `Account::default_address` further strengthens my opinion that `Account` should be framed as "data / metadata about an account" rather than "the inputs needed to construct an account" or "an account identifier". These methods necessarily require embedding the UFVK in the struct (instead of requiring the caller to call `WalletRead::get_unified_full_viewing_keys`), which is independently useful for operating consistently on an account, and I think should be treated as separate from account identification or construction.

Reply:
Yes, the docs were wrong. I've fixed them.